### ISSUE TITLE
SC2154: Incorrect 'referenced but not assigned' warning with :+ expansion

### ISSUE THREAD

**Issue by dimo414** [koalaman/shellcheck#2053]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and found #906 which is related, but a different finding

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
echo "${foo:+"$foo"}"
```

#### Here's what shellcheck currently says:
```
Line 2	SC2154: foo is referenced but not assigned.
```
#### Here's what I wanted or expected to see:

No warning. The inner `"$foo"` will only be evaluated within the context of `:+`, and thus will always be assigned at the time it's referenced.
